### PR TITLE
[SPARK-15782] [YARN] Fix spark.jars and spark.yarn.dist.jars handling

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
## What changes were proposed in this pull request?

When `--packages` is specified with spark-shell the classes from those packages cannot be found, which I think is due to some of the changes in SPARK-12343. 
## How was this patch tested?

Tested manually with both scala 2.10 and 2.11 repls.

@vanzin @davies can you guys please review?

### CODE REVIEW

**Comment by vanzin** [comment]
LGTM since I already tested the 2.10 changes yesterday.

**Comment by vanzin** [comment]
ok to test

**Comment by vanzin** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -2352,6 +2352,26 @@ private[spark] object Utils extends Logging {
...
+   * Unions two comma-separated lists of files and filters out empty strings.
+   */
+  def unionFileLists(leftList: Option[String], rightList: Option[String]): Set[String] = {
+    var allFiles = Set[String]()
+    leftList.foreach { value => allFiles ++= value.split(",") }
+    rightList.foreach { value => allFiles ++= value.split(",") }
+    allFiles.filter { _.nonEmpty }
+  }
+
+  def getUserJars(conf: SparkConf): Seq[String] = {
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: can you add a java doc on this and explain what happens in YARN vs other modes?

**Comment by nezihyigitbasi** [comment]
added javadoc for `getUserJars()`.

**Comment by vanzin** [comment]
Merging to master / 2.0.

**Pull request closed without merging**